webhook/resources/replica/validator.go

```diff
@@ -70,3 +71,51 @@ func (r *replicaValidator) Update(request *admission.Request, oldObj runtime.Obj
 
 	return nil
 }
+
+func (r *replicaValidator) Delete(request *admission.Request, oldObj runtime.Object) error {
+	replica := oldObj.(*longhorn.Replica)
+
+	if err := r.validateReplicaDeletion(replica); err != nil {
+		return werror.NewInvalidError(err.Error(), "")
+	}
+
+	return nil
+}
+
+func (r *replicaValidator) validateReplicaDeletion(replica *longhorn.Replica) error {
+	if replica.Spec.VolumeName == "" {
+		return nil
+	}
+
+	volume, err := r.ds.GetVolumeRO(replica.Spec.VolumeName)
+	if err != nil {
+		if datastore.ErrorIsNotFound(err) {
+			return nil
+		}
+		return errors.Wrapf(err, "failed to get volume %v before deleting replica", replica.Spec.VolumeName)
+	}
+
+	if volume.DeletionTimestamp != nil {
+		return nil
+	}
+
+	replicas, err := r.ds.ListVolumeReplicasRO(volume.Name)
+	if err != nil {
+		return errors.Wrapf(err, "failed to list replicas for volume %v before deleting replica", volume.Name)
+	}
+
+	healthyReplicas := map[string]struct{}{}
+	for _, r := range replicas {
+		if r.Spec.HealthyAt != "" {
```

Comment:
Why not use datastore.IsAvailableHealthreplica() here?

Reply:
`IsAvailableHealthyReplica()` checks failed `failedAt`, but the check is not required in the implementation. explained in https://github.com/longhorn/longhorn-manager/pull/2395#discussion_r1434597274.